pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -190,3 +190,16 @@ func LastErrorWithTaskID(description string, taskID string, codes ...gardencorev
 		},
 	}
 }
+
+// HasNotRetryableErrorCode returns true if at least one of given list of last errors has at least one error code that
+// indicates that an automatic retry would not help fixing the problem.
+func HasNotRetryableErrorCode(lastErrors ...gardencorev1beta1.LastError) bool {
```

Comment:
Just a naming suggestion here, no strong opinion on that one.
```suggestion
func HasNonRetryableErrorCode(lastErrors ...gardencorev1beta1.LastError) bool {
```